File: openmrs-client/src/main/java/org/openmrs/mobile/models/appointment/Provider.java

package org.openmrs.mobile.models.appointment;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Provider {

    @SerializedName("uuid")
    @Expose
    private String uuid;
    @SerializedName("display")
    @Expose
    private String display;
    @SerializedName("person")
    @Expose
    private Person person;
    @SerializedName("identifier")
    @Expose
    private String identifier;
    @SerializedName("attributes")
    @Expose
    private List<Object> attributes = null;
    @SerializedName("retired")
    @Expose
    private Boolean retired;
    @SerializedName("links")
    @Expose
    private List<Link_> links = null;
    @SerializedName("resourceVersion")
    @Expose
    private String resourceVersion;

    public String getUuid() {
        return uuid;
    }

    public void setUuid(String uuid) {
        this.uuid = uuid;
    }

    public String getDisplay() {
        return display;
    }

    public void setDisplay(String display) {
        this.display = display;
    }

    public Person getPerson() {
        return person;
    }

    public void setPerson(Person person) {
        this.person = person;
    }

    public String getIdentifier() {
        return identifier;
    }

    public void setIdentifier(String identifier) {
        this.identifier = identifier;
    }

    public List<Object> getAttributes() {
        return attributes;
    }

    public void setAttributes(List<Object> attributes) {
        this.attributes = attributes;
    }

    public Boolean getRetired() {
        return retired;
    }

    public void setRetired(Boolean retired) {
        this.retired = retired;
    }

    public List<Link_> getLinks() {
        return links;
    }

    public void setLinks(List<Link_> links) {
        this.links = links;
    }

    public String getResourceVersion() {
        return resourceVersion;
    }

    public void setResourceVersion(String resourceVersion) {
        this.resourceVersion = resourceVersion;
    }

}

File: openmrs-android-sdk/src/main/java/com/example/openmrs_android_sdk/library/models/Provider.kt
/*
 * The contents of this file are subject to the OpenMRS Public License
 * Version 1.0 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 * http://license.openmrs.org
 *
 * Software distributed under the License is distributed on an "AS IS"
 * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
 * License for the specific language governing rights and limitations
 * under the License.
 *
 * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
 */

package com.example.openmrs_android_sdk.library.models

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.TypeConverters
import com.example.openmrs_android_sdk.library.models.typeConverters.PersonConverter
import com.example.openmrs_android_sdk.library.models.typeConverters.ProviderAttributeConverter
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.util.*

@Entity(tableName = "provider_table")
class Provider : Resource() {

    @TypeConverters(PersonConverter::class)
    @ColumnInfo(name = "person")
    @SerializedName("person")
    @Expose
    var person: Person? = null

    @ColumnInfo(name = "identifier")
    @SerializedName("identifier")
    @Expose
    var identifier: String? = null

    @TypeConverters(ProviderAttributeConverter::class)
    @ColumnInfo(name = "attributes")
    @SerializedName("attributes")
    @Expose
    var attributes: List<ProviderAttribute> = ArrayList()

    @ColumnInfo(name = "retired")
    @SerializedName("retired")
    @Expose
    var retired: Boolean? = null

    @ColumnInfo(name = "resourceVersion")
    @SerializedName("resourceVersion")
    @Expose
    var resourceVersion: String? = null

}
